### PR TITLE
Ssbffa: setting customs

### PR DESCRIPTION
**Pull request by Arrays-Arr**
had to redo the pr. accidentally did it on my master branch which i updated for another pr

### CODE REVIEW

**Comment by DeathlyPlays** [comment]
Glad someone made these commands!!! HoeenHero and I have been kinda slacking on making them LOL

File: wavelength-plugins/ssb.js [hunk 1]
```diff
@@ -1004,6 +1004,52 @@ exports.commands = {
...
+
+		setcitem: function (target, room, user, connection, cmd, message) {
+			if (!this.can('roomowner')) return false;
+			if (!target) return this.parse('/help ssb setcitem');
+			let targets = target.split(',');
+			let userid = toId(targets[0].trim());
+			if (!userid) return this.errorReply('/help ssb givecitem');
+			let item = toId(targets[1]);
+			if (!item) return this.errorReply('Must include an item');
+			if (!Dex.mod('cssb').getItem(item).exists) return this.errorReply("Item doesnt exist in the ssbffa mod!");
```
Comment (DeathlyPlays) [inline wavelength-plugins/ssb.js]: Doesn't

File: wavelength-plugins/ssb.js [hunk 2]
```diff
@@ -1004,6 +1004,52 @@ exports.commands = {
...
+
+		setcability: function (target, room, user, connection, message) {
+			if (!this.can('roomowner')) return false;
+			if (!target) return this.parse('/help ssb setcability');
+			let targets = target.split(',');
+			let userid = toId(targets[0].trim());
+			if (!userid) return this.errorReply('/help ssb setcability');
+			let customAbility = toId(targets[1]);
+			if (!customAbility) return this.errorReply('/ssb giveability target, ability');
+			if (!Dex.mod('cssb').getAbility(customAbility).exists) return this.errorReply("Ability doesnt exist in the ssbffa mod!");
```
Comment (DeathlyPlays) [inline wavelength-plugins/ssb.js]: Doesn't

File: wavelength-plugins/ssb.js [hunk 3]
```diff
@@ -1004,6 +1004,52 @@ exports.commands = {
...
+
+		setcmove: function (target, room, user, connection, message) {
+			if (!this.can('roomowner')) return false;
+			if (!target) return this.parse('/help ssb setcmove');
+			let targets = target.split(',');
+			let userid = toId(targets[0].trim());
+			if (!userid) return this.parse('/help ssb setcmove');
+			let customMove = toId(targets[1]);
+			if (!customMove) return this.errorReply('Must include a move!');
+			if (!Dex.mod('cssb').getMove(customMove).exists) return this.errorReply("Move doesnt exist in the ssbffa mod!");
```
Comment (DeathlyPlays) [inline wavelength-plugins/ssb.js]: Doesn't

File: wavelength-plugins/ssb.js [hunk 4]
```diff
@@ -1004,6 +1004,52 @@ exports.commands = {
...
+			writeSSB();
+			return this.sendReply('Move set for ' + userid + '!');
+		},
+		setcmovehelp: ['/ssb setcmove targetUser, move'],
+
+		setcability: function (target, room, user, connection, message) {
+			if (!this.can('roomowner')) return false;
+			if (!target) return this.parse('/help ssb setcability');
+			let targets = target.split(',');
+			let userid = toId(targets[0].trim());
```
Comment (HoeenCoder) [inline wavelength-plugins/ssb.js]: redundant

File: wavelength-plugins/ssb.js [hunk 5]
```diff
@@ -1004,6 +1004,52 @@ exports.commands = {
...
+			writeSSB();
+			return this.sendReply('Ability set for ' + userid + '!');
+		},
+		setcabilityhelp: ['/ssb setcmove targetUser, ability'],
+
+		setcitem: function (target, room, user, connection, cmd, message) {
+			if (!this.can('roomowner')) return false;
+			if (!target) return this.parse('/help ssb setcitem');
+			let targets = target.split(',');
+			let userid = toId(targets[0].trim());
```
Comment (HoeenCoder) [inline wavelength-plugins/ssb.js]: redundant

File: wavelength-plugins/ssb.js [hunk 6]
```diff
@@ -1004,6 +1004,52 @@ exports.commands = {
 			}
 		},
 		validatehelp: ['/ssb validate [user] - Validate a users SSBFFA pokemon and if anything invalid is found, set it to its default value. Requires: &, ~'],
+
+		setcmove: function (target, room, user, connection, message) {
+			if (!this.can('roomowner')) return false;
+			if (!target) return this.parse('/help ssb setcmove');
+			let targets = target.split(',');
+			let userid = toId(targets[0].trim());
```
[the remark this answers is not in the document]
Comment (Arrays-Arr) [reply wavelength-plugins/ssb.js]: toId lowercases which is why its used

**Comment by Arrays-Arr** [comment]
I tested this right after the buying check

**Pull request merged**